apps/src/templates/SignInOrAgeDialog.js

```diff
@@ -91,10 +93,16 @@ class SignInOrAgeDialog extends Component {
 
     sessionStorage.setItem(sessionStorageKey, true);
 
-    // Over 13, let them do the tutorial
-    this.setState({
-      open: false
-    });
+    // When opening a new tab, we'll have a new session (and thus show this dialog),
+    // but may still be using a storage_id for a previous user. Clear that cookie
+    // and reload
+    const cookieName = environmentSpecificCookieName('storage_id');
+    if (cookies.get(cookieName)) {
+      cookies.remove(cookieName, {path: '/', domain: '.code.org'});
```

Comment:
This seems to work without the dot before `code.org`, but with the dot seems more right. I tested that this works in Chrome/Safari/FF/IE11.